### PR TITLE
indexer: consistent formula for real time TPS and peak TPS

### PR DESCRIPTION
**Pull request by gegaowp**
## Description 

See title, change real time TPS to be checkpoint based for consistency with peak TPS, also refactor peak TPS query for better readability.

## Test Plan 

- real time TPS: test locally & on devnet by creating testing views and make sure the numbers roughly match with the current formula
- peak TPS: test on devnet by creating a testing view and make sure the numbers match existing view definition

---
If your changes are not user-facing and not a breaking change, you can skip the following section. Otherwise, please indicate what changed, and then add to the Release Notes section as highlighted during the release process.

### Type of Change (Check all that apply)

- [ ] protocol change
- [ ] user-visible impact
- [ ] breaking change for a client SDKs
- [ ] breaking change for FNs (FN binary must upgrade)
- [ ] breaking change for validators or node operators (must upgrade binaries)
- [ ] breaking change for on-chain data layout
- [ ] necessitate either a data wipe or data migration

### CODE REVIEW

File: crates/sui-indexer/migrations/2023-03-20-042226_system_state/up.sql [hunk 1]
```diff
@@ -78,16 +78,42 @@ CREATE TABLE at_risk_validators
...
+  ORDER BY
+    timestamp_ms DESC
+  LIMIT 100
+),
+diff_checkpoints AS (
+  SELECT
+    MAX(sequence_number) as sequence_number,
+    SUM(total_successful_transactions) as total_successful_transactions,
+    timestamp_ms,
+    -(timestamp_ms - LAG(timestamp_ms) OVER (ORDER BY timestamp_ms DESC)) AS time_diff
```
Comment (longbowlu) [inline crates/sui-indexer/migrations/2023-03-20-042226_system_state/up.sql:101]: nit: 
```suggestion
    LAG(timestamp_ms) OVER (ORDER BY timestamp_ms DESC)) - timestamp_ms AS time_diff
```

File: crates/sui-indexer/migrations/2023-03-20-041133_epoch/up.sql [hunk 2]
```diff
@@ -1,24 +1,31 @@
 CREATE MATERIALIZED VIEW epoch_network_metrics as
-SELECT MAX(total_successful_transactions * 1000.0 / time_diff)::float8 as tps_30_days
-FROM (
+WITH checkpoints_30d AS (
```
Comment (longbowlu) [inline crates/sui-indexer/migrations/2023-03-20-041133_epoch/up.sql:2]: how often do we run this query/refresh this view and the recent tps query?

File: crates/sui-indexer/migrations/2023-03-20-041133_epoch/up.sql [hunk 3]
```diff
@@ -1,24 +1,31 @@
...
+  WHERE
+    timestamp_ms > EXTRACT(EPOCH FROM (CURRENT_TIMESTAMP - INTERVAL '30 days')) * 1000
+  GROUP BY
+    timestamp_ms
+),
+tps_data AS (
   SELECT
     sequence_number,
     total_successful_transactions,
     timestamp_ms,
```
Comment (longbowlu) [inline crates/sui-indexer/migrations/2023-03-20-041133_epoch/up.sql:18]: do we need `timestamp_ms` here?

File: crates/sui-indexer/migrations/2023-03-20-042226_system_state/up.sql [hunk 4]
```diff
@@ -78,16 +78,42 @@ CREATE TABLE at_risk_validators
...
+    checkpoints
+  ORDER BY
+    timestamp_ms DESC
+  LIMIT 100
+),
+diff_checkpoints AS (
+  SELECT
+    MAX(sequence_number) as sequence_number,
+    SUM(total_successful_transactions) as total_successful_transactions,
+    timestamp_ms,
```
Comment (longbowlu) [inline crates/sui-indexer/migrations/2023-03-20-042226_system_state/up.sql:100]: ditto

**Review by longbowlu — APPROVED** [review]

File: crates/sui-indexer/migrations/2023-03-20-041133_epoch/up.sql (hunk 2 above)
Comment (gegaowp) [reply crates/sui-indexer/migrations/2023-03-20-041133_epoch/up.sql:2]: this is run every epoch before https://github.com/MystenLabs/sui/pull/12870 and each 100 checkpoints afterwards

**Pull request merged**